src/main/java/seedu/address/model/reservation/Reservation.java

```diff
@@ -0,0 +1,103 @@
+package seedu.address.model.reservation;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Objects;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.model.person.Person;
+
+/**
+ * Represents a Reservation in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class Reservation {
+
+    private final Person person;
+    private final RsvDate date;
+    private final RsvTime time;
+    private final Pax pax;
+
+    /**
+     * Every field must be present and not null.
+     */
+    public Reservation(Person person, RsvDate date, RsvTime time, Pax pax) {
+        requireAllNonNull(person, date, time, pax);
+        this.person = person;
+        this.date = date;
+        this.time = time;
+        this.pax = pax;
+    }
+
+    public Person getPerson() {
+        return this.person;
+    }
+
+    public RsvDate getDate() {
+        return this.date;
+    }
+
+    public RsvTime getTime() {
+        return this.time;
+    }
+
+    public Pax getPax() {
+        return this.pax;
+    }
+
+    /**
+     * Returns true if both reservations are created by the same person,
+     * and have the same date and time.
+     * This defines a weaker notion of equality between two reservations.
+     */
+    public boolean isSameReservation(Reservation otherReservation) {
+        if (otherReservation == this) {
```

Comment:
```suggestion
        if (otherReservation.equals(this)) {
```

Reply:
I think this "==" here just checks if two reservations are the same entity and is sufficient.
I've updated the equals() method after this method to not check for pax.

Reply:
Wait I thought equals() method checks pax but isSameReservation() does not since it is a weaker notion of equality

Reply:
Can refer to the isSamePerson() and equals() in Person class, where
* isSamePerson() is used to enforce no repeated names in the address book, and it only checks name
* equals() is used more in tests / assertions, and it checks every attribute